packages/core-components/src/layout/ErrorBoundary/ErrorBoundary.tsx

```diff
@@ -55,37 +66,61 @@ const SlackLink = (props: { slackChannel?: string | SlackChannel }) => {
   );
 };
 
+const initialState: ErrorBoundaryState = {
+  error: null,
+};
+
 /** @public */
-export const ErrorBoundary: ComponentClass<
-  ErrorBoundaryProps,
-  State
-> = class ErrorBoundary extends Component<ErrorBoundaryProps, State> {
+export class ErrorBoundary extends Component<
+  PropsWithChildren<ErrorBoundaryProps>,
+  ErrorBoundaryState
+> {
+  static getDerivedStateFromError(error: Error) {
+    return { error };
+  }
+
   constructor(props: ErrorBoundaryProps) {
     super(props);
-    this.state = {
-      error: undefined,
-      errorInfo: undefined,
-    };
+
+    this.resetErrorBoundary = this.resetErrorBoundary.bind(this);
+    this.state = initialState;
   }
 
   componentDidCatch(error: Error, errorInfo: ErrorInfo) {
     // eslint-disable-next-line no-console
     console.error(`ErrorBoundary, error: ${error}, info: ${errorInfo}`);
-    this.setState({ error, errorInfo });
+
+    this.props.onError?.(error, errorInfo);
+  }
+
+  resetErrorBoundary(...args: any[]) {
```

Comment:
Is there some use case for the args that you're envisioning right now? Could perhaps just leave resets as zero-argument operations unless there is a solid intended usage you have in mind.

Reply:
@freben None of this came from my mind/vision. Everything that you've referenced to change (`args`, `null` -> `undefined`, the type name `FallbackProps`) was copied from the [source implementation](https://github.com/bvaughn/react-error-boundary). I don't mind changing them. I'm just clarifying why they are the way they are.

Reply:
Yepyep I gatherer as much, no worries in that regard. Things are looking good and it's mostly tweaks to fit my mindset and the repo style / philosophy a bit.

Reply:
We can leave the `args` I guess even though I don't think we really have a use case right now. Naming etc would be good to get tweaked though as per above comments